extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -150,29 +152,73 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
-						value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
-						validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+						label: constants.connectionProfile
+					},
+					{
+						label: constants.userConnectionString
 					}
-				) ?? '';
-
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
+				];
+				const selectedConnectionString = await vscode.window.showQuickPick(connectionStringOptions, {
+					canPickMany: false,
+					title: constants.selectConnectionString,
+					ignoreFocusOut: true
+				});
+				if (!selectedConnectionString) {
+					// User cancelled
+					return;
 				}
-
-				try {
-					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, newConnectionStringValue);
-					if (success) {
-						connectionStringSettingName = newConnectionStringSettingName;
+				if (selectedConnectionString.label === constants.userConnectionString) {
+					const newConnectionStringValue = await vscode.window.showInputBox(
+						{
+							title: constants.enterConnectionString,
+							ignoreFocusOut: true,
+							value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
+							validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+						}
+					) ?? '';
+					if (!newConnectionStringValue) {
+						// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+						continue;
+					} else {
+						continue;
+					}
+				} else {
+					// get connection string for sql server
+					let connectionString: string = '';
+					let connectionUri: string = '';
+					const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+					if (!connectionInfo) {
+						// User cancelled
+						return;
+					}
+					try {
+						// TO DO: https://github.com/microsoft/azuredatastudio/issues/18012
+						connectionUri = await vscodeMssqlApi.connect(connectionInfo);
+					} catch (e) {
+						// give an error if unable to connect to selected connection and go back to select setting quickpick
+						console.warn(e);
+						void vscode.window.showErrorMessage(utils.getErrorMessage(e));
+						continue;
+					}
+					try {
+						connectionString = await vscodeMssqlApi.getConnectionString(connectionUri, false);
+						try {
+							const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, connectionString);
+							if (success) {
+								connectionStringSettingName = newConnectionStringSettingName;
+							}
+						} catch (e) {
+							// display error message and show select setting quickpick again
+							void vscode.window.showErrorMessage(utils.getErrorMessage(e));
```

Comment:
For consistency should add a continue here as well

And same comment about adding a prefix to the error `Failed to set connection string app setting: <ERROR>`

Reply:
see comment above